src/Reflection/ClassReflection.php

```diff
@@ -1047,6 +1047,11 @@ public function isImmutable(): bool
 			$this->isImmutable = $resolvedPhpDoc !== null && $resolvedPhpDoc->isImmutable();
 		}
 
+		$parentClass = $this->getParentClass();
+		if ($parentClass !== null && !$this->isImmutable) {
```

Comment:
Please move everything into the `if ($this->isImmutable === null) {` code block - the whole logic of this method is that the property caches the result. The computation should be done inside that code block.